File: HW/HW2/BadVersion.py
def bad_version(n):
    index = n/2+1
    if (isBadVersion(index)):
        while isBadVersion(index-1):
            index -= 1
        return int(index)
    else:
        while not isBadVersion(index):
            index += 1
        return int(index)

def isBadVersion(version):
    if (version>=2):
        return True
    else:
        return False

print(bad_version(2))
print(bad_version(6))

File: HW/HW2/BadVersion.java
public class BadVersion {
    public static int bad_version(int n){
        int index = n/2+1;
        if (isBadVersion(index)){
            while (isBadVersion(index-1)){
                index--;
            }
            return index;
        }
        else {
            while (!isBadVersion(index)){
                index++;
            }
            return index;
        }
    }
    public static boolean isBadVersion(int version){
        if (version>=5){
            return true;
        }
        return false;
    }
    public static void main(String[] args){
        System.out.println(bad_version(3));
        System.out.println(bad_version(11));
    }
}